x/twap/types/utils.go

```diff
@@ -14,8 +14,17 @@ func NewTwapRecord(k AmmInterface, ctx sdk.Context, poolId uint64, denom0 string
 	if !(denom0 > denom1) {
```

Comment:
ah we actually need a new stateful constructor, to get `lastErrTime` from the last TWAP record

Reply:
Done